src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotRemoveBaseTypeOrInterfaceTests.cs

```diff
@@ -0,0 +1,294 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class CannotRemoveBaseTypeOrInterfaceTests
+    {
+        [Fact]
+        public void PromotedBaseClassOrInterfaceIsNotReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First : FirstBase, IFirstInterface { }
+  public class Second : SecondBase { }
+  public class SecondBase { }
+  public class FirstBase { }
+  public interface IFirstInterface { }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First : NewBase { }
+  public class Second : NewSecondBase { }
+  public class NewBase : FirstBase, INewInterface { }
+  public class FirstBase { }
+  public class SecondBase { }
+  public class NewSecondBase : NewSecondBaseBase { }
+  public class NewSecondBaseBase : SecondBase { }
+  public interface IFirstInterface { }
+  public interface INewInterface : IFirstInterface { }
+}
+";
+
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+
+            ApiComparer differ = new();
+
+            Assert.Empty(differ.GetDifferences(left, right));
+        }
+
+        [Fact]
+        public void RemovedInterfaceAndBaseClassAreReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First : FirstBase, IFirstInterface { }
+  public class FirstBase { }
+  public interface IFirstInterface { }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First : NewBase { }
+  public class NewBase : INewInterface { }
+  public class FirstBase { }
+  public interface IFirstInterface { }
+  public interface INewInterface { }
+}
+";
+
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+
+            ApiComparer differ = new();
+
+            CompatDifference[] differences = differ.GetDifferences(left, right).ToArray();
+
+            CompatDifference[] expected = new[]
+            {
+                new CompatDifference(DiagnosticIds.CannotRemoveBaseType, string.Empty, DifferenceType.Changed, "T:CompatTests.First"),
+                new CompatDifference(DiagnosticIds.CannotRemoveBaseInterface, string.Empty, DifferenceType.Changed, "T:CompatTests.First"),
+            };
+
+            Assert.Equal(expected, differences, CompatDifferenceComparer.Default);
+
+            differences[0].Message.Contains("CompatTests.FirstBase");
```

Comment:
any specific reason on here we are testing the messages as well ?

Reply:
They are localized and that can cause tests to fail based on the culture settings of the computer executing the tests.